compiler/rustc_errors/src/diagnostic_builder.rs

```diff
@@ -99,20 +99,22 @@ impl<'a> DerefMut for DiagnosticBuilder<'a> {
 
 impl<'a> DiagnosticBuilder<'a> {
     /// Emit the diagnostic.
-    pub fn emit(&mut self) {
+    pub fn emit(&mut self) -> ErrorReported {
         self.0.handler.emit_diagnostic(&self);
         self.cancel();
+        ErrorReported::new_i_know_what_im_doing()
```

Comment:
This (and everything else in this file) doesn't need the unchecked constructor, we're in `rustc_errors`.